src/Facade/Services/ConsignmentShipfileFacadeService.cs

```diff
@@ -71,5 +76,31 @@ public IResult<ConsignmentShipfile> DeleteShipfile(int id)
             this.repository.Remove(toDelete);
             return new SuccessResult<ConsignmentShipfile>(toDelete);
         }
+
+        public IResult<ConsignmentShipfile> Add(ConsignmentShipfileResource resource)
+        {
+            if (!int.TryParse(resource.InvoiceNumbers, out var invoiceNo))
+            {
+                return new BadRequestResult<ConsignmentShipfile>("Invalid Invoice Number");
+            }
+
+            var consignment =
+                this.consignmentRepository.FindBy(x => x.Invoices.Any(i => i.DocumentNumber == invoiceNo));
+
+            if (consignment == null)
+            {
+                return new BadRequestResult<ConsignmentShipfile>("Consignment Not Found");
+            }
+
+            var shipfile = new ConsignmentShipfile
+                               {
+                                   ConsignmentId = consignment.ConsignmentId,
+                                   Id = this.repository.FindAll().Max(x => x.Id) + 1
```

Comment:
do we need to specify an id for shipfiles? I haven't seen us have to do this ^ before, just seen us setting an Id for ones that use the get_next_val function or whatever (doesn't mean we don't need to here though)

Reply:
Using a sequence is a better idea anyway